### PR TITLE
Update for 4.0 release

### PR DESCRIPTION
**Pull request by ardalis**

### CODE REVIEW

**Comment by ardalis** [comment]
@maxkoshevoi let's try this again. I was mistakenly using an older branch that should have been deleted. I updated the README to add a note - does that cover what you meant?

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ An endpoint that previously inherited from the synchronous `BaseEndpoint` should
 +     .WithActionResult<IEnumerable<WeatherForecast>>
 ```
 
-The above change typically would not require any change to the `Handle` method. Endpoints that inherited from `BaseAsyncEndpoint` would now use `EndpointBaseAsync`.
+The above change typically would not require any change to the `Handle` method. Endpoints that inherited from `BaseAsyncEndpoint` would now use `EndpointBaseAsync`. You can also just inherit from `EndpointBaseAsync` directly (without the `.With*` additions) which will provide you with a controller with a single Handle method, if you need more flexibility than the fluent generic interface provides.
```
Comment (maxkoshevoi) [inline README.md:41]:
```suggestion
The above change typically would not require any change to the `Handle` method. Endpoints that inherited from `BaseAsyncEndpoint` would now use `EndpointBaseAsync`. You can also just inherit from `EndpointBase` directly (without the `.With*` additions) which will provide you with a controller with a single Handle method without restrictions on parameter amount and type, if you need more flexibility than the fluent generic interface provides.
```

File: src/Ardalis.ApiEndpoints/Ardalis.ApiEndpoints.csproj [hunk 2]
```diff
@@ -11,8 +11,8 @@
     <Description>An alternative to Controllers for ASP.NET Core API Endpoints.</Description>
     <Summary>Controllers promote creating bloated classes that lack cohesion. This project provides a simpler alternative that follows SOLID principles. An alternative to Controllers for ASP.NET Core API Endpoints.</Summary>
     <PackageTags>aspnet asp.net aspnetcore asp.net core api web api rest endpoint controller</PackageTags>
-    <PackageReleaseNotes>TODO</PackageReleaseNotes>
-    <Version>4.0.0-pre1</Version>
+    <PackageReleaseNotes>See README for breaking changes and migration guide.</PackageReleaseNotes>
+    <Version>4.0.0</Version>
     <AssemblyName>Ardalis.ApiEndpoints</AssemblyName>
     <PackageLicenseFile>LICENSE.txt</PackageLicenseFile>
   </PropertyGroup>
```
Comment (maxkoshevoi) [inline src/Ardalis.ApiEndpoints/Ardalis.ApiEndpoints.csproj:18]: Update the analyzer dependency as well

Should be:
```
<PackageReference Include="Ardalis.ApiEndpoints.CodeAnalyzers" Version="4.0.0" />
```

**Review by maxkoshevoi — APPROVED** [review]
Looks great!

**Comment by ardalis** [comment]
Thanks! Let's (finally) get this live!

**Comment by ardalis** [comment]
Now if only the CI build(s) would work...

**Comment by maxkoshevoi** [comment]
> Now if only the CI build(s) would work...

I think, you just need to install .net 6 here:
https://github.com/ardalis/ApiEndpoints/blob/main/.github/workflows/codeql-analysis.yml

**Comment by ardalis** [comment]
Oh probably. That fixed the other ones but I wasn't worried about codeql last night.

**Pull request merged**